Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -1104,38 +1104,26 @@ int CLuaEngineDefs::EngineGetVisibleTextureNames(lua_State* luaVM)
     return 1;
 }
 
-int CLuaEngineDefs::EngineSetModelVisibleTime(lua_State* luaVM)
+bool CLuaEngineDefs::EngineSetModelVisibleTime(const std::variant<std::string, unsigned int> variantModelID, char hourOn, char hourOff)
 {
-    // bool engineSetModelVisibleTime ( int/string modelID, int hourOn, int hourOff )
-    SString strModelId;
-    char cHourOn,cHourOff;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadString(strModelId);
-    argStream.ReadNumber(cHourOn);
-    argStream.ReadNumber(cHourOff);
+    if (hourOn > 24 || hourOn < 0)
+        throw std::invalid_argument("hour on must be between 0 and 24");
 
-    if (!argStream.HasErrors())
-    {
-        ushort      usModelID = CModelNames::ResolveModelID(strModelId);
-        CModelInfo* pModelInfo = g_pGame->GetModelInfo(usModelID);
-        if (pModelInfo)
-        {
-            if (cHourOn > cHourOff)
-                std::swap(cHourOn, cHourOff);
+    if (hourOff > 24 || hourOff < 0)
+        throw std::invalid_argument("hour on must be between 0 and 24");
 
-            if (cHourOn >= 0 && cHourOn <= 24 && cHourOff >= 0 && cHourOff <= 24)
-            {
-                lua_pushboolean(luaVM, pModelInfo->SetTime(cHourOn, cHourOff));
-                return 1;
-            }
-        }
-    }
-    else
-        luaL_error(luaVM, argStream.GetFullErrorMessage());
+    if (hourOn > hourOff)
+        std::swap(hourOn, hourOff);
 
-    // Failed
-    lua_pushboolean(luaVM, false);
-    return 1;
+
+    // either convert string to modelid using CModelNames::ResolveModelID, or assign the id from the variant
+    const unsigned short modelID = std::holds_alternative<std::string>(variantModelID) ? CModelNames::ResolveModelID(std::get<std::string>(variantModelID)) : std::get<unsigned int>(variantModelID);
```

Comment:
EngineSetModelVisibleTime

```suggestion
    unsigned short modelID;
    if (std::holds_alternative<std::string>(variantModelID))
        modelID = CModelNames::ResolveModelID(std::get<std::string>(variantModelID));
    else
        modelID = std::get<unsigned int>(variantModelID);
```